src/hooks/useSearchHighlightAndScroll.ts

```diff
@@ -3,43 +3,51 @@ import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
 import type {SearchQueryJSON} from '@components/Search/types';
 import type {ReportActionListItemType, ReportListItemType, SelectionListHandle, TransactionListItemType} from '@components/SelectionList/types';
 import * as SearchActions from '@libs/actions/Search';
+import {isReportActionEntry} from '@libs/SearchUIUtils';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import type {SearchResults, Transaction} from '@src/types/onyx';
+import type {ReportActions, SearchResults, Transaction} from '@src/types/onyx';
 import usePrevious from './usePrevious';
 
 type UseSearchHighlightAndScroll = {
     searchResults: OnyxEntry<SearchResults>;
     transactions: OnyxCollection<Transaction>;
     previousTransactions: OnyxCollection<Transaction>;
+    reportActions: OnyxCollection<ReportActions>;
+    previousReportActions: OnyxCollection<ReportActions>;
     queryJSON: SearchQueryJSON;
     offset: number;
 };
 
 /**
- * Hook used to trigger a search when a new transaction is added and handle highlighting and scrolling.
+ * Hook used to trigger a search when a new transaction or report action is added and handle highlighting and scrolling.
  */
-function useSearchHighlightAndScroll({searchResults, transactions, previousTransactions, queryJSON, offset}: UseSearchHighlightAndScroll) {
+function useSearchHighlightAndScroll({searchResults, transactions, previousTransactions, reportActions, previousReportActions, queryJSON, offset}: UseSearchHighlightAndScroll) {
     // Ref to track if the search was triggered by this hook
     const triggeredByHookRef = useRef(false);
     const searchTriggeredRef = useRef(false);
     const previousSearchResults = usePrevious(searchResults?.data);
     const [newSearchResultKey, setNewSearchResultKey] = useState<string | null>(null);
-    const highlightedTransactionIDs = useRef<Set<string>>(new Set());
+    const highlightedIDs = useRef<Set<string>>(new Set());
     const initializedRef = useRef(false);
+    const isChat = queryJSON.type === CONST.SEARCH.DATA_TYPES.CHAT;
 
-    // Trigger search when a new transaction is added
+    // Trigger search when a new report action is added while on chat or when a new transaction is added for the other search types.
     useEffect(() => {
         const previousTransactionsLength = previousTransactions && Object.keys(previousTransactions).length;
         const transactionsLength = transactions && Object.keys(transactions).length;
 
-        // Return early if search was already triggered or there's no change in transactions length
-        if (searchTriggeredRef.current || previousTransactionsLength === transactionsLength) {
+        const reportActionsLength = reportActions && Object.values(reportActions).reduce((sum, curr) => sum + Object.keys(curr ?? {}).length, 0);
```

Comment:
I think this can be changed to this, no?

```suggestion
        const reportActionsLength = reportActions && Object.values(reportActions).length;
```

And also below

Reply:
Nope @marcochavezf the structure of reportActions is 
```
{
reportAction_reportID:{
reportActionID:{},
reportActionID2:{}
...
}
}
```

Reply:
Oh cool, thanks for the clarification!